keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -73,636 +85,560 @@
+
+    // Protocols is the set of protocols the provisioned Bucket supports. BucketAccesses can request
+    // to access this BucketClaim using any of the values given here.
+    // Possible values: S3, Azure, GCS
+    Protocols []Protocol
+
+    // Error holds the most recent error message, with a timestamp.
+    // This is cleared when provisioning is successful.
+    Error *TimestampedError
   }
 ```
-## BucketClass
 
-Resouce for configuring common properties for multiple Buckets. BucketClass is cluster-scoped.
+#### BucketClass
+
+Resource for configuring common properties for multiple Buckets. BucketClass is cluster-scoped.
 
 ```go
 BucketClass {
   TypeMeta
   ObjectMeta
 
-  // DriverName is the name of driver associated with this bucket
+  // DriverName is the name of driver associated with this bucket.
+  // +required
   DriverName string
 
-  // DeletionPolicy is used to specify how COSI should handle deletion of this
-  // bucket. There are 3 possible values:
-  //  - Retain: Indicates that the bucket should not be deleted from the OSP
-  //  - Delete: Indicates that the bucket should be deleted from the OSP
-  //        once all the workloads accessing this bucket are done
+  // DeletionPolicy determines whether a Bucket created through the BucketClass should be deleted
+  // when its bound BucketClaim is deleted.
+  // Possible values:
+  //  - Retain: keep the Bucket object as well as the bucket in the underlying storage system
+  //  - Delete: delete the Bucket object as well as the bucket in the underlying storage system
+  // +required
   DeletionPolicy DeletionPolicy
 
-  // Parameters is an opaque map for passing in configuration to a driver
-  // for creating the bucket
+  // Parameters is an opaque map for passing in configuration to a driver for creating the bucket.
   // +optional
   Parameters map[string]string
 }
 ```
 
-## BucketAccess
+#### BucketAccess
+
+The BucketAccess is used to request access to a bucket. It contains fields for choosing the Bucket for which the credentials will be generated, and also includes a bucketAccessClassName field, which in-turn contains configuration for authorizing users to access buckets.
 
-A resource to access a Bucket. BucketAccess is namespace-scoped
+A resource to access a Bucket. BucketAccess is namespace-scoped.
+
+For many OSP drivers, each BucketAccess will correspond with a unique OSP backend 'user' (e.g., an S3 user). COSI suggests this as a starting point, but this is not a strict requirement.
+
+COSI end Users should generally expect that BucketAccess-BucketClaim pairings should not result in BucketAccesses that are not able to read or write unrelated BucketClaims. Stated another way, BucketAccess A referencing BucketClaim A should generally not be able to read or write to BucketClaim B. COSI drivers may choose to have options for different operating modes to support niche scenarios, but this should be the default assumption for BucketAccess provisioning.
```

Comment:
What does `operating modes` mean here?  Does that refer to access mode such as read-write and read-only?

Reply:
Replaced "operating modes" with "behaviors" here to make it more clear that it's a generalized statement about non-specific behaviors.